### PR TITLE
Resolve audit result

### LINKED ISSUE #458: Resolve audit result values
#### Currently...

* `result.value` - most of the time is actually === a score
* `result.rawValue` - many times its a string of "1232.32ms"  .. not exactly "raw"

#### Proposal 
* `result.score`
* `result.rawValue` - which is either a number or boolean, never a string
*  `result.displayValue` - which is a string

### PR DESCRIPTION
**Pull request by wardpeet**
Should fix #458 
No idea if I did any good but basically this is what i've done.

Every value is mapped to rawValue (so value => rawValue).
Score is equal to score, if no score is set we fallback to rawValue
display value uses rawvalue with minor adaptation like adding ms or fallbacks to string representation of rawValue. If displayValue === score we make it empty so we don't display it in our reporting

### CODE REVIEW

File: lighthouse-core/audits/audit.js [hunk 1]
```diff
@@ -29,12 +29,24 @@ class Audit {
    * @return {!AuditResult}
    */
   static generateAuditResult(result) {
-    if (typeof result.value === 'undefined') {
-      throw new Error('generateAuditResult requires a value');
+    if (typeof result.rawValue === 'undefined') {
+      throw new Error('generateAuditResult requires a rawValue');
+    }
+
+    let score = result.score ? result.score : result.rawValue;
```
Comment (paulirish) [inline lighthouse-core/audits/audit.js]: should probably use an undefined check as to not discard possible scores of 0

Comment (wardpeet) [inline lighthouse-core/audits/audit.js]: correct :)

**Comment by paullewis** [comment]
Looks like it fails tests :(

**Comment by wardpeet** [comment]
Forgot to push :'( waiting for the build :)

**Comment by paulirish** [comment]
awesome stuff. A massive cleanup. Thanks for this

**Pull request merged**